(NO-DPS) Web docs for Chat API (#1459)
livechat/livechat-public-docs: src/pages/messaging/chat-api/index.mdx

---
weight: 80
category: "messaging"
title: Overview
tagline: "Build messaging software with Chat API by Text."
desc: "API for a two-way communication between users. Available as a Web API."
---

# Introduction

Chat API by Text is a backend service that allows for a two-way communication between users using the HTTP protocol (Web reference).

Chat API won't let you customize the LiveChat application, but it will let you build your own chat application from scratch.

## Features

- Communication can happen on public channels or via direct messages

> - Storing messaging history

Reviewer: What's the difference between this and "users can get previous messages from the channels they have joined"? I'm asking as this is described like there's a bigger difference than just "users can see their message history" but I might be wrong :D

Reviewer 2: I think it's about seeing messages that were sent when you were not yet part of a channel, but I'd have to confirm with @kacperf531

Reviewer 3: yeah, that's right, I think the former description says it more explicitly. Currently there is no mechanism to prevent users from getting messages in the channel before they joined. But maybe it's obvious? If that's the case please let me know so I'll update the description to the latter version

Reviewer 2: No problem, we can present it as a feature, not a limitation :) I don't think it's obvious, it's just that "storing message history" and "users can get previous messages..." seemed very similar at the first glance. I rephrased it a little and now I think the difference is more visible

> - Users can get previous messages from the channels they have joined

Reviewer 2: ```suggestion
- Users can see messages that were sent before they joined the channel
```
